File: js/11729.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const plates = Number(input[0]);
let cnt = 0;
let ans = '';
function countingHanoi(n, from, to, temp) {
  if(!n) return;
  countingHanoi(n-1, from, temp, to);
  ans += from + ' ' + to + '\n';
  cnt++;
  countingHanoi(n-1, temp, to, from);
}

countingHanoi(plates,1,3,2);
console.log(cnt);
console.log(ans);

File: python/11729.py
def hanoi(k, f, t, tmp) :
  global ans
  global cnt
  if k == 0 : return
  hanoi(k-1, f, tmp, t)
  ans.append(str(f) + ' ' + str(t))
  cnt += 1
  hanoi(k-1, tmp, t, f)
  
n = int(input())
cnt = 0
ans = list()

hanoi(n,1,3,2)
print(cnt)
print(*ans, sep="\n")
